pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -509,6 +509,16 @@ public PositionImpl getMaxReadPosition() {
         }
     }
 
+    @Override
+    public CompletableFuture<Position> getLastDispatchablePosition() {
+        PositionImpl tnxMaxReadPosition = getMaxReadPosition();
+        if (tnxMaxReadPosition.compareTo((PositionImpl) topic.getLastPosition()) == 0) {
+            return topic.getLastDispatchablePosition();
+        } else {
+            return CompletableFuture.completedFuture(tnxMaxReadPosition);
+        }
+    }
```

Comment:
If we want to introduce this method in the TransactionBuffer. I think we'd better make sure the behavior is correct. The aborted transactions is not dispatchable, so we'd filter them out.

Or we can just revert this change and open another PR to handle the transaction issues which will be more better approach because the PR title said it fixes the issue related to replicate subscription state.

Reply:
+1